From: Andrew Morton <akpm@linux-foundation.org>
Date: Mon, 10 Aug 2026 20:50:56 -0700
Subject: [PATCH] lib/Kconfig.debug: make DEBUG_FORCE_WEAK_PER_CPU depend on
 S390||ALPHA

CONFIG_DEBUG_FORCE_WEAK_PER_CPU appears to be alpha-specific.  The Kconfig
help text says s390 can need it also, but the comment in percpu-defs.h
only mentions alpha.  The original commit 7c756e6e19e7 ("percpu: implement
optional weak percpu definitions") mentions both alpha and s390.

All quite mysterious and perhaps this feature is no longer needed.

However x86_64 allmodconfig enables it, and because mm/alloc_tag.c
requires !DEBUG_FORCE_WEAK_PER_CPU (another mystery), alloc_tag isn't
getting allmodconfig build coverage.

Adding `depends on S390||ALPHA' fixes this.

Cc: Tejun Heo <tj@kernel.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 lib/Kconfig.debug | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index b82515cde538c..7d1cfc5ed2d37 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -616,6 +616,7 @@ config BUILTIN_MODULE_RANGES
 config DEBUG_FORCE_WEAK_PER_CPU
 	bool "Force weak per-cpu definitions"
 	depends on DEBUG_KERNEL
+	depends on S390 || ALPHA
 	help
 	  s390 and alpha require percpu variables in modules to be
 	  defined weak to work around addressing range issue which
-- 
2.47.3

